### DISCUSSION TITLE
How to get trace data with langfuse-langchain in JS/TS?

### DISCUSSION THREAD

**Discussion by AdiletBekmuratov** [langfuse/langfuse#2005] [Support] [answered]
I searched all documentation and only found how to create and update traces. Is there any way to get a trace after its creation via langfuse-langchain in JS/TS SDK? I'm able to get trace ID with `langfuseHandler.traceId`

"langfuse-langchain": "^3.9.0"

**Comment by marcklingen** [comment]
what do you want to do with the trace? currently there is no get by id function in js but you can always just use the rest api directly: https://api.reference.langfuse.com

**Reply by AdiletBekmuratov** [reply]
I want to get usage (tokens, cost) and save it in my working db for each request. Langchain's usage count doesn't work properly in cases of agents and tools used, so I decided to get the tokens count from Langfuse.

**Comment by marcklingen** [comment, marked as answer]
We have recently added methods to the JS SDK to cover this use case: https://langfuse.com/docs/query-traces
